File: backend/src/main/java/com/studyai/wellness/dto/StatsDto.java
package com.studyai.wellness.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/**
 * Data Transfer Object for Statistics information.
 *
 * <p>This DTO aggregates various statistics including overview metrics,
 * weekly stats, achievements, and goals.</p>
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class StatsDto {

    /**
     * List of overview metrics.
     */
    private List<MetricDto> overview;

    /**
     * List of weekly statistics.
     */
    private List<WeeklyStatDto> weeklyStats;

    /**
     * List of user achievements.
     */
    private List<AchievementDto> achievements;

    /**
     * List of user goals.
     */
    private List<GoalDto> goals;
}

File: android/app/src/main/java/com/studyai/wellness/data/model/StatsDto.kt
package com.studyai.wellness.data.model

data class StatsDto(
    val overview: List<MetricDto>,
    val weeklyStats: List<WeeklyStatDto>,
    val achievements: List<AchievementDto>,
    val goals: List<GoalDto>
)

data class WeeklyStatDto(
    val label: String,
    val value: Double,
    val target: Double
)

data class AchievementDto(
    val id: String,
    val title: String,
    val description: String,
    val icon: String,
    val unlockedAt: String? = null
)

data class GoalDto(
    val id: String,
    val title: String,
    val current: Double,
    val target: Double,
    val unit: String,
    val deadline: String
)
